src/RequestHandlerSwooleRunner.php

```diff
@@ -81,43 +114,88 @@ function (Throwable $exception) use ($serverRequestErrorResponseGenerator) : Res
-                $request->server['request_method'],
-                $request->server['request_uri']
-            );
-            $emit = new SwooleEmitter($response);
-            try {
-                $psr7Request = ($this->serverRequestFactory)($request);
-            } catch (Throwable $e) {
-                // Error in generating the request
-                $this->emitMarshalServerRequestException($emit, $e);
-                return;
-            }
-            $emit->emit($this->handler->handle($psr7Request));
-        });
+        $this->swooleHttpServer->on('start', [$this, 'onStart']);
+        $this->swooleHttpServer->on('request', [$this, 'onRequest']);
         $this->swooleHttpServer->start();
     }
 
+    /**
+     * On start event for swoole http server
+     */
+    public function onStart(SwooleHttpServer $server) : void
+    {
+        printf("Swoole is running at %s:%s\n", $server->host, $server->port);
+    }
+
+    /**
+     * On HTTP request event for swoole http server
+     */
+    public function onRequest(
+        SwooleHttpRequest $request,
+        SwooleHttpResponse $response
+    ) {
+        printf(
+            "%s - %s - %s %s\n",
+            date('Y-m-d H:i:sO', time()),
+            $request->server['remote_addr'],
+            $request->server['request_method'],
+            $request->server['request_uri']
+        );
+        if ($this->getStaticResource($request, $response)) {
+            return;
+        }
+        $emit = new SwooleEmitter($response);
+        try {
+            $psr7Request = ($this->serverRequestFactory)($request);
+        } catch (Throwable $e) {
+            // Error in generating the request
+            $this->emitMarshalServerRequestException($emit, $e);
+            return;
+        }
+        $emit->emit($this->handler->handle($psr7Request));
+    }
+
+    /**
+     * Emit marshal server request exception
+     */
     private function emitMarshalServerRequestException(
         EmitterInterface $emitter,
         Throwable $exception
     ) : void {
         $response = ($this->serverRequestErrorResponseGenerator)($exception);
         $emitter->emit($response);
     }
+
+    /**
+     * Get a static resource, if any, and set the swoole HTTP response
+     */
+    private function getStaticResource(
+        SwooleHttpRequest $request,
+        SwooleHttpResponse $response
+    ) : bool {
+        $staticFile = $this->docRoot . $request->server['request_uri'];
+        if (! isset($cacheTypeFile[$staticFile])) {
+            if (! file_exists($staticFile)) {
+                return false;
+            }
+            $type = pathinfo($staticFile, PATHINFO_EXTENSION);
+            if (! isset($this->allowedStatic[$type])) {
+                return false;
+            }
+            $cacheTypeFile[$staticFile] = $this->allowedStatic[$type];
+        }
+        $response->header('Content-Type', $cacheTypeFile[$staticFile]);
+        $response->sendfile($staticFile);
```

Comment:
Do you know if files are gzipped by default? It would be nice to have gzip compression support and possiblity to align the compression level.

Reply:
According to the documentation [Swoole\HTTP\Response::sendfile](https://www.swoole.co.uk/docs/modules/swoole-http-server/methods-properties#swoole-http-response-sendfile) doesn't support gzip. We can think to support it in a future PR.